snuba/query/processors/type_converters/uuid_array_column_processor.py

```diff
@@ -0,0 +1,47 @@
+import uuid
+
+
+from snuba.query.expressions import (
+    Column,
+    Expression,
+    FunctionCall,
+    Literal,
+    Lambda,
+    Argument,
+)
+from snuba.query.processors.type_converters import BaseTypeConverter, ColumnTypeError
+
+
+class UUIDArrayColumnProcessor(BaseTypeConverter):
+    def _translate_literal(self, exp: Literal) -> Expression:
+        try:
+            assert isinstance(exp.value, str)
+            new_val = str(uuid.UUID(exp.value))
+            return FunctionCall(exp.alias, "toUUID", (Literal(None, value=new_val),))
+        except (AssertionError, ValueError):
+            raise ColumnTypeError("Not a valid UUID string")
+
+    def _process_expressions(self, exp: Expression) -> Expression:
+        if isinstance(exp, Column) and exp.column_name in self.columns:
+            return FunctionCall(
```

Comment:
Couple edge cases that I don't think this is correctly handling.

1) If there is something like `arraySlice(hierarchical_hashes, 0, 2)` this will put the `arrayMap` inside the `arraySlice`, so we translate the entire array instead of just the slice. Not a huge deal but it would be nice if we could only translate the slice.

2) We should spend some time trying to figure out how this works if someone adds `hierarchical_hashes` to the arrayJoin clause. I'm not sure if Clickhouse will handle that cleanly.

Reply:
1. fixed 
2. I'm not entirely sure what is expected in the case of arrayJoin (I don't have a usecase for it) but I added a testcase that demonstrates current behavior

Reply:
I just wanted to make sure that if someone does an arrayjoin on hierarchical_hashes that the processor won't break the query. I think this makes sense.